src/gi.js

```diff
@@ -1,12 +1,20 @@
 import g from "./bindings/mod.js";
 import handleInfo from "./handleInfo.js";
 
+const repos = new Map();
+
 /**
  * @param {string} namespace
  * @param {string?} version
  * @returns
  */
 export function require(namespace, version) {
+  const key = `${namespace}-${version ?? ""}`;
```

Comment:
creating key this way is not suitable.
if version is not passed, latest version of library available on the system will be loaded, so for example
```js
require("GLib");
require("GLib", "2.0");
```
should be the same.

Reply:
> if version is not passed, latest version of library available on the system will be loaded, so for example

If the version is not passed, the latest version will always be loaded, which makes the cache return the latest version and the behaviour is correct. What we can do is like GJS, and warn if there are multiple library versions.

Maybe to make the code correct, we can list all the versions and automatically load the latest one.

```
gjs> imports.gi.Gtk
typein:1:1 warning: Requiring Gtk but it has 2 versions available; use imports.gi.versions to pick one
[object GIRepositoryNamespace]
```

Reply:
I've updated the code to handle the above scenario, please take a look if you can.

Reply:
LGTM. Thanks.